### PR TITLE
Bug fix for using addDays with negative numbers so that it results in…

### PR DESCRIPTION
**Pull request by dg76**
… a previous month.

### CODE REVIEW

**Comment by dnfield** [comment]
It would be really nice if we could have some tests for this - for now, maybe even just in the example app would work.

**Comment by dg76** [comment]
I have added a few tests now.

File: test/gregorian_calendar_test.dart [hunk 1]
```diff
@@ -10,6 +10,11 @@ void main() {
     expect(endOfJan.addDays(10), equals(new GregorianCalendar(2018, 2, 10)));
     expect(endOfJan.addDays(28), equals(new GregorianCalendar(2018, 2, 28)));
     expect(endOfJan.addDays(29), equals(new GregorianCalendar(2018, 3, 1)));
+
+    final GregorianCalendar startOfSept = new GregorianCalendar(2019, 9, 2);
+    expect(startOfSept.addDays(-1), equals(new GregorianCalendar(2019, 9, 1)));
```
Comment (dnfield) [inline test/gregorian_calendar_test.dart:15]: Ahh nice, I forgot I had tests in here. This looks great, thanks!

**Review by dnfield — APPROVED** [review]
LGTM

**Comment by dg76** [comment]
Great, thank you for integrating the fix! :-)

**Pull request merged**